ansible_collection_helper.py

```diff
@@ -7,6 +7,53 @@
 from pathlib import Path
 from typing import Set
 
+import yaml
+from ansible.cli.doc import DocCLI
+from jinja2 import Environment, FileSystemLoader
+
+env = Environment(loader=FileSystemLoader((Path(__file__).parent / "templates")))
+template = env.get_template("readme.jinja2")
+
+
+def generate_role_documentation(
+    role_name: str, role_spec_file: Path, molecule_dir: Path, doc_dir: Path
+) -> None:
+    """Reads roles metadata and molecule converge playbooks to generate README
+    files for a specific role."""
+    playbook = molecule_dir / role_name / "converge.yml"
+    doc = doc_dir / f"role_{role_name}_doc.md"
+    with (
+        open(role_spec_file, "r") as f_spec,
+        open(playbook, "r") as f_play,
+        open(doc, "w") as f_doc,
```

Comment:
Since you have `pathlib.Path` objects, use their `.open()` method:
```suggestion
        role_spec_file.open() as f_spec,
        playbook.open() as f_play,
        doc.open("w") as f_doc,
```